docs/building/forms.md

```diff
@@ -0,0 +1,107 @@
+A form is like a collection of menus or raw responses and it is basically collecting information from the user.
+
+It is composed by a **header**, a **body** and a **footer**. The body represents form items and these can be slightly modified menus or raw responses.
+
+The form items are being processed in a successive manner. After all the form items have been processed, the user can confirm his choices or reset any of them.
+
+## Json Structure
+
+Your server must return a JSON response with a structure similar to the one below:
+
+```
+{'body': [{'type': 'string',
+           'name': 'descr',
+           'description': 'Provide a description',
+           'method': 'POST',
+           'path': '/callback-path'},
+          {'type': 'date'
+           'name': 'due_date',
+           'header': 'due date',
+           'description': 'Provide a due date'},
+          {'type': 'menu',
```

Comment:
From the previous 2 items I understand that `type` specifies what data type should be returned by Nautilus (to the dev app) for a specific wizard step. If this is true, then the value `menu` here breaks that logic. Here, `type` seems to specify what should Nautilus render to the user (a menu). Am I wrong?

Reply:
Its a matter of convention. The type on the first level indicates whether its a form or menu. The type in the forms body indicates the wizard step type. Menu refers to an options wizard step, while others are self explanatory.

I've tried to reuse the menu structure as much as I could.